build_scripts/build_runner.py

```diff
@@ -822,30 +798,44 @@ def main():
     try:
         if not args.changed_repo and not args.repo_states:
             log.error('"--changed-repo" or "--repo-states" argument bust be added')
-            exit(Error.CRITICAL.value)
+            exit(ErrorCode.CRITICAL.value)
         elif args.changed_repo and args.repo_states:
             log.warning('The --repo-states argument is ignored because the --changed-repo is set')
 
-        build_config.generate_build_config()
-        build_config.run_stage(args.stage)
-    except Exception as exc:
-        if not isinstance(exc, subprocess.CalledProcessError):
-            log.exception("Exception occurred")
-        log.error("%sING FAILED", args.stage.name)
-        build_state_file = pathlib.Path(args.root_dir) / 'build_state'
-        build_state_file.write_text(json.dumps({'status': "FAIL"}))
-        exit(Error.CRITICAL.value)
+        # prepare build configuration
+        if build_config.generate_build_config():
+            # run stage of build
+            no_errors = build_config.run_stage(args.stage)
+        else:
+            log.critical('Can not generate build configuration')
+            no_errors = False
+
+    except Exception:
+        no_errors = False
+        log.exception('Exception occurred')
 
+    build_state_file = pathlib.Path(args.root_dir) / 'build_state'
+    if no_errors:
+        if not build_state_file.exists():
+            build_state_file.write_text(json.dumps({'status': "PASS"}))
+        log.info('-' * 50)
+        log.info("%sING COMPLETED", args.stage.name)
+    else:
+        build_state_file.write_text(json.dumps({'status': "FAIL"}))
+        log.error('-' * 50)
+        log.error("%sING FAILED", args.stage.name)
+        exit(ErrorCode.CRITICAL.value)
 
 if __name__ == '__main__':
     if platform.python_version_tuple() < ('3', '6'):
         print('\nERROR: Python 3.6 or higher required')
-        exit(Error.CRITICAL.value)
+        exit(1)
```

Comment:
why not Error.CRITICAL.value

Reply:
Error class lies in helper. It does not import when python version < 3.6